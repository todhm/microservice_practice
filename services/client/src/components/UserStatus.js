import React,{Component} from 'react';
import axios from 'axios';
import {Link} from 'react-router-dom';
export default class UserStatus extends Component{
    state={
        email:'',
        id:'',
        username:''
    };

    componentDidMount(){
        this.getUserStatus();
    }

    getUserStatus=(event)=>{
        const options = {
            url: `${process.env.REACT_APP_USERS_SERVICE_URL}/auth/status`,
            method: 'get',
            headers:{
                'Content-Type': 'application/json',
                Authorization: `Bearer ${window.localStorage.authToken}`
            }
        };
        return axios(options)
            .then((res)=>{
                const{email,id,username} = res.data.data; 
                this.setState({email,id,username})
            })
            .catch((error)=>{console.log(error.response.data);})
    }

    render(){
        const {id,email,username} = this.state; 
        const {isAuthenticated} = this.props; 
        if(!isAuthenticated){
            return(
                <p>You must logged in to view this. Click <Link to="/login">here </Link>to log back in</p>
            )
        }

        return(
            <div>
                <ul>
                    <li><strong>User ID:</strong>{id}</li>
                    <li><strong>Email:</strong>{email}</li>
                    <li><strong>Username:</strong>{username}</li>
                </ul>
            </div>
        )
    };

}