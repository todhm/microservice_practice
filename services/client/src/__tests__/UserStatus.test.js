import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import UserStatus from "../components/UserStatus";
import {MemoryRouter as Router} from 'react-router-dom';

const authenticatedList = [true, false];

beforeAll(()=>{
  global.localStorage={
    getItem:()=>'someToken',
  };
});
authenticatedList.forEach((auth)=>{
    test(` render properly when auth is ${String(auth)}`, () => {
        const wrapper = shallow(<UserStatus isAuthenticated={auth}/>);
      });
      
})

test("UsersList renders a snapshot properly", () => {
  const tree = renderer.create(<Router><UserStatus isAuthenticated={true}/></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
