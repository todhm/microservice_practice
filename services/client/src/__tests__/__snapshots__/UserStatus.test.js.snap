// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <ul>
    <li>
      <strong>
        User ID:
      </strong>
      
    </li>
    <li>
      <strong>
        Email:
      </strong>
      
    </li>
    <li>
      <strong>
        Username:
      </strong>
      
    </li>
  </ul>
</div>
`;
